firestore/tests/unit/v1/test_document.py

```diff
@@ -735,6 +735,7 @@ def test_non_existent(self):
         snapshot = self._make_one(None, None, False, None, None, None)
         as_dict = snapshot.to_dict()
         self.assertIsNone(as_dict)
+        self.assertIsNone(snapshot.id)
```

Comment:
Was it actually tested, in both cases, to see where it fails?

Reply:
it tests if there is not_existent document, so the document id will be None. It will not throw error.

You can show error throws cause of this line  in issue 7564.
```python
print(list(db.get_all([db.collection("something").document('thisDoesNotExists')]))[0].id)
```